mlflow/models/model_config.py

```diff
@@ -11,8 +11,56 @@
 
 class ModelConfig:
     """
-    ModelConfig used in code to read a YAML configuration file, and this configuration file can be
-    overridden when logging a model.
+    ModelConfig used in code to read a YAML configuration file or a dictionary.
+    When invoking the ModelConfig locally in a model file, development_config can be passed in
+    which would be used as configuration for the model. But this development_config configuration
+    file will be overridden when logging a model.
+
+    Args:
+        development_config: Path to the YAML configuration file or a dictionary containing the
+                            configuration. If the configuration is not provided, an error is raised.
+
+    .. code-block:: python
+        :caption: Example usage in model code
+
+        from mlflow.models import ModelConfig
+
+        # Load the configuration from a file
+        config = ModelConfig(development_config="config.yaml")
```

Comment:
Can you add an example of `config.yaml`? It's unclear what this YAML file should look like.